src/components/AddComment.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import { PostComment } from '../../api-post';
+import '../Styling/Addcomment.css'
+export const AddComment = ({ articleId }) => {
+    const [loading, setLoading] = useState(false);
+
+    const username = 'weegembump';
+
+    const handleCommentSubmit = (event) => {
+        event.preventDefault();
+        setLoading(true);
+
+        const form = event.target;
+        const body = form.elements.body.value;
+
+        const comment = {
+            username,
+            body,
+        };
+
+        PostComment(articleId, comment)
```

Comment:
This isn't a component so shouldn't be capitalised please

Reply:
So there's a slight problem in using window.location in react, as it relates directly to the DOM and therefore any updates aren't reflected in our state. How could we trigger a react re-render instead?

Reply:
This is also not making the most of Optimistic Rendering.  At the moment, we're waiting for the request to complete before re-setting the comments. We want this to optimistically show the update before the request completes, and then revert the change if it fails. Take a look at the notes on Optimistic Rendering for more info.